File: Java/Digit/program24.java
//Accept the number from user and check whether the number is armstrong or not

import java.util.*;

class Digit
{
    public int iNo;

    public Digit(int x)
    {
        iNo = x;
    }

    public void Display()
    {
        System.out.println("The number is: "+this.iNo);
    }

    public int Power(int iNo1,int iNo2)
    {
        int iCnt = 0;
        int iMult = 1;

        for(iCnt = 1; iCnt <= iNo2; iCnt++)
        {
            iMult = iMult * iNo1;
        }
        return iMult;
    }

    public boolean CheckArmstrong()
    {
        int iDigit = 0, iDigCnt = 0;
        int iTemp = 0, iSum = 0;

        iTemp = iNo;

        while(iNo != 0)
        {
            iDigCnt++;
            iNo = iNo / 10;
        }

        iNo = iTemp;

        while(iNo != 0)
        {
            iDigit = iNo % 10;
            iSum = iSum + Power(iDigit,iDigCnt);
            iNo = iNo / 10;
        }

        if(iSum == iTemp)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

class program24
{
    public static void main(String arg[])
    {
        int iValue = 0;
        boolean bRet;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the number");
        iValue = sobj.nextInt();

        Digit obj = new Digit(iValue);

        obj.Display();

        bRet = obj.CheckArmstrong();
        if(bRet == true)
        {
            System.out.println("The number is armstrong number");
        }
        else
        {
            System.out.println("The number is not armstrong number");
        }
    }
}

File: C++/Number/program24.cpp
// check whether the number is even or not

#include <iostream>
using namespace std;

class Number
{
    public:
        int iNo;

        Number(int x)
        {
            iNo = x;
        }

        void Display()
        {
            cout<<"The numbers is: "<< this->iNo <<endl;
        }

        bool ChkEven()
        {   
            if(iNo % 2 == 0)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
};

int main()
{
    int iValue = 0;
    bool bRet;

    cout << "Enter the number" << endl;
    cin >> iValue;

    Number nobj(iValue);

    nobj.Display();

    bRet = nobj.ChkEven();
    if(bRet == true)
    {
        cout<<"The enterd number is even"<<endl;
    }
    else
    {
        cout<<"The enterd number is odd"<<endl;
    }
    

    return 0;
}
